src/config/mongoose_config_validator.erl

```diff
@@ -11,7 +11,8 @@
 -type validator() ::
         any | non_empty | non_negative | positive | module | {module, Prefix :: atom()}
       | jid | domain | domain_template | url | ip_address | ip_mask | network_address | port
-      | filename | dirname | loglevel | pool_name | shaper | access_rule | {enum, list()}.
+      | filename | dirname | loglevel | pool_name | shaper | access_rule | {enum, list()}
+      | {domain_template, string()}.
```

Comment:
Minor: the order here is getting more out of sync with the order of clauses in `validate/3`... but I guess it would require some shuffling to make it the same anyway. I think this is OK as it is.